--- log.go.old	2021-12-02 03:21:04.903318398 -0800
+++ log.go	2021-12-02 03:24:41.532034480 -0800
@@ -1,6 +1,6 @@
 package log
 
-import "fmt"
+import "os"
 
 type Level uint8
 
@@ -34,12 +34,25 @@
 	return ""
 }
 
+func writeString(str string) error {
+	var offset int
+	for offset < len(str) {
+		n, err := os.Stdout.WriteString(str[offset:])
+		if err != nil {
+			return err
+		}
+		offset += n
+	}
+	return nil
+}
+
 func logOut(level Level, msg string) {
 	if level <= curLvl {
 		if level != curLvl {
-			fmt.Print(level.name(), ": ")
+			writeString(level.name())
+			writeString(": ")
 		}
-		fmt.Println(msg)
+		writeString(msg)
 	}
 }
 
